consensus/blake3pow/consensus.go

```diff
@@ -456,6 +456,15 @@ func (blake3pow *Blake3pow) verifyHeader(chain consensus.ChainHeaderReader, head
 			return fmt.Errorf("invalid baseFee: have %s, want %s, parentBaseFee %s, parentGasUsed %d",
 				expectedBaseFee, header.BaseFee(), parent.BaseFee(), parent.GasUsed())
 		}
+		// Verify that the stateUsed is <= stateLimit
+		if header.StateUsed() > header.GasLimit() {
```

Comment:
Did you mean to check header.StateUsed() against header.StateLimit() ?

Reply:
oh yeah, good catch

Reply:
Changed